app/services/authentication_service.rb

```diff
@@ -0,0 +1,5 @@
+module AuthenticationService
+  def self.authenticate_user(params)
+    User.find_by(buck_id: params[:buck_id])&.authenticate(params[:password])
```

Comment:
This is susceptible to [timing attacks](https://en.wikipedia.org/wiki/Timing_attack) if you're worried about it.